x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -43,168 +49,189 @@ function ActionMenuButton({ onClick }: { onClick: () => void }) {
   );
 }
 
+function removeEmptyItems<T extends {}>(
+  items: Array<T | null | undefined>
+): T[] {
+  return items.filter(item => !!item) as T[];
+}
```

Comment:
Why not keep the inline filter on the `condition`?

Reply:
TypeScript doesn't understand what filter does (AFAICT) without a little help. This is a little nudge for TypeScript to understand that there will not be any null or undefined items in the array. Let's see what it looks like after I use `condition` everywhere though.

Reply:
No longer necessary 👍